lib/custom_set_cases.rb

```diff
@@ -0,0 +1,85 @@
+class CustomSetCase < OpenStruct
+  def name
+    'test_%s' % description.gsub(/ |-/, '_')
+  end
+
+  def test_body
+    send section
```

Comment:
I was trying to attack this vector from the other end... This is much simpler... Thanks @Insti !